server/server.go

```diff
@@ -473,49 +478,51 @@ func (s *server) CreateBlockchains(ctx context.Context, req *rpcpb.CreateBlockch
 	}
 
 	s.log.Debug("CreateBlockchains")
-	if info := s.getClusterInfo(); info == nil {
+	info := s.getClusterInfo()
+	if info == nil {
+		s.mu.Unlock()
 		return nil, ErrNotBootstrapped
 	}
 
 	if len(req.GetBlockchainSpecs()) == 0 {
+		s.mu.Unlock()
 		return nil, ErrNoBlockchainSpec
 	}
 
 	chainSpecs := []network.BlockchainSpec{}
 	for _, spec := range req.GetBlockchainSpecs() {
 		chainSpec, err := getNetworkBlockchainSpec(s.log, spec, false, s.network.pluginDir)
 		if err != nil {
+			s.mu.Unlock()
 			return nil, err
 		}
 		chainSpecs = append(chainSpecs, chainSpec)
 	}
 
 	// check that defined subnets exist
 	subnetsMap := map[string]struct{}{}
-	for _, subnet := range s.clusterInfo.Subnets {
+	for _, subnet := range info.Subnets {
 		subnetsMap[subnet] = struct{}{}
 	}
 	for _, chainSpec := range chainSpecs {
 		if chainSpec.SubnetId != nil {
 			_, ok := subnetsMap[*chainSpec.SubnetId]
 			if !ok {
+				s.mu.Unlock()
 				return nil, fmt.Errorf("subnet id %q does not exits", *chainSpec.SubnetId)
 			}
 		}
 	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
 	// if there will be a restart, network will not be healthy
 	// until finishing
 	for _, chainSpec := range chainSpecs {
 		if chainSpec.SubnetId == nil {
-			s.clusterInfo.Healthy = false
+			info.Healthy = false
 		}
 	}
-
-	s.clusterInfo.CustomChainsHealthy = false
+	info.CustomChainsHealthy = false
+	s.setClusterInfo(info)
```

Comment:
Also, you are setting the info here. Why does setting not have a normal write lock, and instead only a read lock? With a read lock only, some other go routine could be writing to clusterInfo concurrently?